### PR TITLE
Add new hubs to MegaMenu

### PR DESCRIPTION
**Pull request by ncksllvn**
## Description
This PR updates the design system to receive the updates for handling hubs without child pages in the design system PR below.

Content PR - https://github.com/department-of-veterans-affairs/vagov-content/pull/335
Design system PR - https://github.com/department-of-veterans-affairs/veteran-facing-services-tools/pull/48

## Testing done
- Comparing local vs Staging in different screen sizes

## Screenshots
See design system PR, https://github.com/department-of-veterans-affairs/veteran-facing-services-tools/pull/48

## Acceptance criteria
- [x] New hubs are added into the mega menu
- [x] Nothing broken

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Comment by 1Copenut** [comment]
I merged my `<MegaMenu />` changes into the master branch, but have not cut a new release to NPM. Change PR here: https://github.com/department-of-veterans-affairs/veteran-facing-services-tools/pull/28

I also merged into vets-website the groundwork CSS to ensure the `<MegaMenu />` update doesn't break the visual layout. Change PR here: https://github.com/department-of-veterans-affairs/vets-website/pull/9804

LTM if I should go ahead and cut an NPM release and issue a new PR for pulling my changes into vets-website, or you will capture them both with this PR.

File: src/site/stages/build/plugins/create-header-footer.js [hunk 1]
```diff
@@ -59,8 +59,10 @@ function createHeaderFooterData(buildOptions) {
     // property, has to exist in a separate file for editing/organizational reasons.
     // To accomplish this, we borrow use the fragments concept. More info in that middleware.
     for (const hub of vaBenefitsAndHealthCare.menuSections) {
-      applyFragments(buildOptions, metalsmith, hub.links);
-      delete hub.links.fragments;
+      if (hub.links && hub.links.fragments) {
+        applyFragments(buildOptions, metalsmith, hub.links);
+        delete hub.links.fragments;
+      }
```
Comment (ncksllvn) [inline src/site/stages/build/plugins/create-header-footer.js]: The new hubs won't have a `links` property so we have to check for this now.

File: src/platform/site-wide/sass/modules/_m-vet-nav.scss [hunk 2]
```diff
@@ -452,3 +456,84 @@ body.va-pos-fixed {
...
+      box-shadow: 0 5px 9px -3px $color-base;
+    }
+  }
+
+  // Toggle button on mobile
+  [class^="vetnav-controller"] {
+    position: relative !important;
+  }
+
+}
```
Comment (ncksllvn) [inline src/platform/site-wide/sass/modules/_m-vet-nav.scss]: These styles were just copied over from the merger shame.

File: src/platform/site-wide/sass/merger.scss [hunk 3]
```diff
@@ -559,10 +493,6 @@ $medium-phone-screen: 375px;
   }
 }
 
-[class^="vetnav-controller"] {
-  position: relative !important;
-}
-
```
Comment (ncksllvn) [inline src/platform/site-wide/sass/merger.scss]: Moved all this vetnav stuff into vet-nav SCSS.

**Comment by ncksllvn** [comment]
To review -

1. Pull this branch
2. Pull the `conten-wip-add-hubs-to-top-nav` branch in `vagov-content`
3. `yarn install` in `vets-website` to update Formation locally
4. Start the site

**Comment by 1Copenut** [comment]
I've tried running this a couple of times. When I run the commands listed above, the text appears in the mega menu, but they're buttons instead of links. The text is there and if i manually change it to a link, styling is correct.

Not sure it's related, but the content PR shows a failing test: https://github.com/department-of-veterans-affairs/vagov-content/pull/335

If I'm doing something wrong, I'll revise and re-check within the hour.

Screen Shot 2019-03-28 at 1 15 29 PM

**Comment by ncksllvn** [comment]
@1Copenut In your screenshot, then new hubs have arrow pointing right. Those shouldn't be there. I think something is wrong with your Yalc configuration?

The failing test in vagov-content is expected. This PR has to be merged for that to pass.

**Review by jbalboni — APPROVED** [review]
This looks good to me, but I can't test it locally currently.

**Comment by 1Copenut** [comment]
@ncksllvn I'm not sure what I was doing wrong, still haven't figured it out, but the PR looks good to me as well.

**Pull request merged**